src/cow-react/modules/trade/hooks/useTradeTypeInfo.ts

```diff
@@ -0,0 +1,34 @@
+import { matchPath, useLocation } from 'react-router-dom'
+import { Routes } from '@cow/constants/routes'
+import { match } from 'react-router'
+import { TradeStateFromUrl } from '@cow/modules/trade/types/TradeState'
+import { useMemo } from 'react'
+
+export enum TradeType {
+  SWAP,
+  LIMIT_ORDER,
+}
+
+export interface TradeTypeInfo {
+  tradeType: TradeType
+  route: Routes
+  match: match<TradeStateFromUrl>
+}
+
+export function useTradeTypeInfo(): TradeTypeInfo | null {
+  const location = useLocation()
+
+  const swapMatch = matchPath<TradeStateFromUrl>(location.pathname, Routes.SWAP)
+  const limitOrderMatch = matchPath<TradeStateFromUrl>(location.pathname, Routes.LIMIT_ORDER)
+
+  return useMemo(() => {
+    if (!swapMatch && !limitOrderMatch) return null
+
+    return {
+      tradeType: swapMatch ? TradeType.SWAP : TradeType.LIMIT_ORDER,
+      route: swapMatch ? Routes.SWAP : Routes.LIMIT_ORDER,
```

Comment:
Is it possible to have tradeType != route?

Reply:
No, because above we check:
```
if (!swapMatch && !limitOrderMatch) return null
```